src/support/mtx_rw.c

```diff
@@ -219,12 +225,13 @@ __wt_readlock(WT_SESSION_IMPL *session, WT_RWLOCK *l)
          * If we are the first reader to queue, set the next read group. Note: don't re-read from
          * the lock or we could race with a writer unlocking.
          */
-        new.u.v = old.u.v;
+        __wt_atomic_storev64(&new.u.v, old.u.v);
         if (new.u.s.readers_queued++ == 0)
             new.u.s.reader = new.u.s.next;
         ticket = new.u.s.reader;
         WT_ASSERT(session, new.u.s.readers_queued != 0);
-        if (__wt_atomic_casv64(&l->u.v, old.u.v, new.u.v))
+        if (__wt_atomic_casv64(
```

Comment:
I'm surprised we need to do this but I guess it is how our cas is written.

Reply:
Fixed in [0242fdf](https://github.com/wiredtiger/wiredtiger/pull/10236/commits/0242fdf671db575063c6d91211871d941b5dd8af)